packages/core/src/utils/toArray.ts

```diff
@@ -0,0 +1,12 @@
+import type { ProofRequest } from '@aries-framework/core'
+
+export function attributesToArray(proofRequest: ProofRequest) {
+  return Array.from(proofRequest.requestedAttributes.values()).reduce<string[]>(
+    (names, a) => [...names, ...(a.name ? [a.name] : a.names ? a.names : [])],
+    []
+  )
+}
+
+export function predicatesToArray(proofRequest: ProofRequest) {
+  return Array.from(proofRequest.requestedPredicates.values()).map((a) => a.name)
+}
```

Comment:
This is also very proof request specific, so maybe we should host all the utils (assertDuplicates,  toArray) in a util file called `indyProofRequest.ts` (either in the root util directory, or the proofs specific util directory)